src/main/java/app/api/InformationController.java

```diff
@@ -53,9 +68,76 @@ public String getLastBlock() {
                 .limit(1).iterator();
 
         return cursor.hasNext() ? cursor.next().toJson() : "{}";
-
     }
 
+   	@RequestMapping(value = ApiPaths.PRODUCER_YIELD, method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseBody
+    public String getProducerYield() {
+ 
+    	final Integer blocksPerHour = 7200;
+    	final Long witnessNum = 21L;
+       	final Float maxBlocksPerHour = blocksPerHour/ (witnessNum *1.0f);
+   		String localProducer = "";
+    	Long producedBlocks = 0L;
+    	
+    	try {
+    	
+            final MongoCollection<Document> collection = mongoClient.getDatabase("application")
+            		.getCollection("data");
+
+            if(collection.countDocuments() > 0L) {
+            	Document applicationData = collection.find().first();
+            	if(! applicationData.getString("localNodePrivKey").equals("")) {
+            		ECPrivateKey privKey = cryptoService.getECPrivateKeyFromRawString((String) applicationData.getString("localNodePrivKey"));
+   	   				localProducer = cpxkey.getPublicAddressCPX(privKey).toString();
+            	}
+            }
+                 		
+    		final HashMap<String, Long> producerStats = new HashMap<>();
+    		final MongoCursor<Document> cursorMiner = mongoClient.getDatabase("apex")
+                    .getCollection("miner")
+                    .find().sort(new Document("addr", -1))
+                    .limit(0).iterator();
+    		
+    		while (cursorMiner.hasNext()) {
+                producerStats.put((String) cursorMiner.next().get("addr"),0L);
+    		}
+
+            final MongoCursor<Document> cursor = mongoClient.getDatabase("apex")
+               		 .getCollection("block")
+               		 .find(gte("timeStamp", new BsonDateTime(Instant.now().toEpochMilli() - 3600000L)))   
+               		 .limit(0)
+               		 .iterator();
+            
+            if(cursor.hasNext()) {
+            	cursor.forEachRemaining(entry -> {  
+            		if(entry.get("confirmed").toString().equals("true")) {
+            			Long newVal = (Long) producerStats.get(entry.get("producer")) + 1L;
+            			producerStats.replace(entry.get("producer").toString(), newVal);
+            		}
+            	});
+            }   
+            if(! localProducer.equals("") && producerStats.containsKey(localProducer))
+            	producedBlocks = producerStats.get(localProducer);	
+ 
+    	} catch(Exception e) {
+        	e.printStackTrace();
+        }
+        
+    	Float yield = (producedBlocks *1.0f / maxBlocksPerHour) * 100.0f;
+    	String formattedString = String.format("%.01f", yield);
+    	
+        final HashMap<String, Object> entry = new HashMap<>();
+        entry.put("producerYield", formattedString + "% (" + producedBlocks + " blocks/hour)");
+        try {
+            return jacksonWriter.getStringFromRequestObject(entry);
+        } catch (JsonProcessingException e) {
+            return "[]";
```

Comment:
You return a HashMap normally so maybe change this to "{}" for empty HashMap instead of List